### PR TITLE
Add QCD scale, alpha strong and PDF shape uncertainties

### PR DESCRIPTION
**Pull request by bfonta**
**Description**:

This PR includes the variables for QCD scale, PDF and alpha strong uncertainties. The GEN POG recommended way requires the propagation of all variables via a histogram, totalling 110 histograms, which must later on be converted into 6 histograms (three uncertainties times 2: the up and down variations). 

The corrections are sample dependent; I've encoded this in the ```datasets_UL*.txt``` files.

**Missing Samples**:

Some MiniAOD samples seem to be missing, and recovery from tape might be needed. I've compiled an incomplete list of the problematic samples [here](https://codimd.web.cern.ch/8pLwr7xYRq-Fj4iD796cbA?both).
For those samples I had to *guess* the weight scheme, based on similarly-named samples; the choice might be wrong.

#### Testing C++ code

I've tested one sample with the "Powheg9" weight scheme with 10 events:

```
11-Oct-2024 13:23:25 CEST  Successfully opened file root://xrootd-cms.infn.it//store/mc/RunIISummer20UL18MiniAODv2/ZH_HToBB_ZToLL_M-125_TuneCP5_13TeV-powheg-pythia8/MINIAODSIM/106X_upgrade2018_realistic_v16_L1v1-v1/40000/CB1759C2-7F0B-0041-837E-40EF92EF73D7.root
Unable to load sec.protocol plugin libXrdSecztn.so
Begin processing the 1st record. Run 1, Event 1645528, LumiSection 4253 on stream 0 at 11-Oct-2024 13:23:29.282 CEST
0.120101 0.173682
Begin processing the 2nd record. Run 1, Event 1645526, LumiSection 4253 on stream 0 at 11-Oct-2024 13:23:29.284 CEST
0.0729855 0.409239
Begin processing the 3rd record. Run 1, Event 1645527, LumiSection 4253 on stream 0 at 11-Oct-2024 13:23:29.285 CEST
0.078123 1.11333
Begin processing the 4th record. Run 1, Event 1645531, LumiSection 4253 on stream 0 at 11-Oct-2024 13:23:29.286 CEST
0.167114 0.0868412
Begin processing the 5th record. Run 1, Event 1645532, LumiSection 4253 on stream 0 at 11-Oct-2024 13:23:29.287 CEST
0.128011 0.120531
Begin processing the 6th record. Run 1, Event 1645534, LumiSection 4253 on stream 0 at 11-Oct-2024 13:23:29.287 CEST
0.124584 0.118244
Begin processing the 7th record. Run 1, Event 1645529, LumiSection 4253 on stream 0 at 11-Oct-2024 13:23:29.288 CEST
0.113043 0.518984
Begin processing the 8th record. Run 1, Event 1645530, LumiSection 4253 on stream 0 at 11-Oct-2024 13:23:29.289 CEST
0.0625943 0.157486
Begin processing the 9th record. Run 1, Event 1645533, LumiSection 4253 on stream 0 at 11-Oct-2024 13:23:29.290 CEST
0.305153 0.158264
Begin processing the 10th record. Run 1, Event 1645525, LumiSection 4253 on stream 0 at 11-Oct-2024 13:23:29.291 CEST
0.0538194 0.142008
11-Oct-2024 13:23:29 CEST  Closed file root://xrootd-cms.infn.it//store/mc/RunIISummer20UL18MiniAODv2/ZH_HToBB_ZToLL_M-125_TuneCP5_13TeV-powheg-pythia8/MINIAODSIM/106X_upgrade2018_realistic_v16_L1v1-v1/40000/CB1759C2-7F0B-0041-837E-40EF92EF73D7.root
```

where the first number is the PDF uncertainty and the second is the QCD scale uncertainty. 
On the KLUB side we will store these numbers in new branches (adding 1 given what ```combine``` expected), and will propagate them until the final histogram shapes.

#### Testing full submission

Tested with a 2018 TTTo2L2Nu sample.

image

### CODE REVIEW

File: NtupleProducer/plugins/HTauTauNtuplizer.cc [hunk 1]
```diff
@@ -1486,11 +1489,12 @@ void HTauTauNtuplizer::beginJob(){
...
-    myTree->Branch("MC_weight_scale_muF2",&_MC_weight_scale_muF2,"MC_weight_scale_muF2/F");
-    myTree->Branch("MC_weight_scale_muR0p5",&_MC_weight_scale_muR0p5,"MC_weight_scale_muR0p5/F");
-    myTree->Branch("MC_weight_scale_muR2",&_MC_weight_scale_muR2,"MC_weight_scale_muR2/F");
-    myTree->Branch("MC_weight_PSWeight0",&_MC_weight_PSWeight0,"MC_weight_PSWeight0/F");
+
+	myTree->Branch("MC_QCDscale", &_MC_QCDscale, "MC_QCDscale/F");
+	myTree->Branch("MC_pdf", &_MC_pdf, "MC_pdf/F");
+	myTree->Branch("MC_astrong", &_MC_astrong, "MC_astrong/F");
+
+	myTree->Branch("MC_weight_PSWeight0",&_MC_weight_PSWeight0,"MC_weight_PSWeight0/F");
```
Comment (dzuolo) [inline NtupleProducer/plugins/HTauTauNtuplizer.cc:1599]: Question originating from my ignorance: will we need these PSWeights with the new implementation of the uncertainty scheme?

Comment (bfonta) [reply NtupleProducer/plugins/HTauTauNtuplizer.cc:1599]: Not AFAIK, but the communication via CMS Talk is so far extremely hard to follow, so I cannot guarantee.

Comment (dzuolo) [reply NtupleProducer/plugins/HTauTauNtuplizer.cc:1599]: Ok, thanks for checking! And how did we end up with 100 new branches? What do they contain?

Comment (bfonta) [reply NtupleProducer/plugins/HTauTauNtuplizer.cc:1599]: We end up with new branches as follows:
- 101 for the PDF variations
- 7 for the QCD variations
- 2 for the tow alpha strong variations
So in total 108 new branches. The nominal is given by the weight branch we already have.

Comment (dzuolo) [reply NtupleProducer/plugins/HTauTauNtuplizer.cc:1599]: So we need to add 50 systematic uncertainties to the analysis, just for the PDFs... ?

Comment (bfonta) [reply NtupleProducer/plugins/HTauTauNtuplizer.cc:1599]: We only have to add 3 systematics, but they require all the 110 branches above. Details were clarified in private.
